packages/csvviewer-extension/src/searchprovider.ts

```diff
@@ -4,13 +4,17 @@ import { ISearchProvider, ISearchMatch } from '@jupyterlab/documentsearch';
 import { CSVViewer } from '@jupyterlab/csvviewer';
 import { IDocumentWidget, DocumentWidget } from '@jupyterlab/docregistry';
 import { Signal, ISignal } from '@phosphor/signaling';
+import { Widget } from '@phosphor/widgets';
 
-export class CSVSearchProvider implements ISearchProvider {
+// The type for which canSearchFor returns true
+export type CSVDocumentWidget = IDocumentWidget<CSVViewer>;
+
+export class CSVSearchProvider implements ISearchProvider<CSVDocumentWidget> {
   /**
    * Report whether or not this provider has the ability to search on the given object
    */
-  static canSearchOn(domain: any): boolean {
-    // check to see if the CMSearchProvider can search on the
+  static canSearchOn(domain: Widget): domain is CSVViewer {
```

Comment:
This is the wrong type (`domain.content` is `CSVViewer`). Ideally, this should have caused compile / tests to fail.
```suggestion
  static canSearchOn(domain: Widget): domain is IDocumentWidget<CSVViewer> {
```

Reply:
Since it passes compile, there's nothing to test for once the typescript is gone. Is there a way to tip off the compiler? I'll see if I can tweak `ISearchProviderConstructor` and how that interface is consumed.

Reply:
Also, there's an issue highlighted here. Should the signature of `CSVSearchProvider` actually be:

```typescript
export class CSVSearchProvider implements ISearchProvider<DocumentWidget<CSVViewer>> {
```

since the `canSearchOn` method is testing for the concrete class  `DocumentWidget<CSVViewer>`, not the interface `IDocumentWidget<CSVViewer>`?

Reply:
The domain type is fixed in the latest.

Though I couldn't figure out a way to make the compiler fail on the signature with the wrong domain type.

Reply:
If there is no simple way to get it to fail compilation, it is probably ok to leave it as it is, as long as it has the correct typing now.

Reply:
> Should the signature of CSVSearchProvider actually be ...

If the search provider only supports `DocumentWidget`, and not the generic `IDocumentWidget` (on account of the `instanceof`), I guess that makes sense. I'll leave the decision to you.

Reply:
If some hypothetical renegade coder came by and implemented another class that `implements IDocumentWidget`, `CSVSearchProvider` would not work correctly with it. So `DocumentWidget<CSVViewer>` does make more sense to me, and that's what I'll go with.